modules/apps/portal-workflow/portal-workflow-kaleo-runtime-impl/src/main/java/com/liferay/portal/workflow/kaleo/runtime/internal/petra/executor/GraphWalkerPortalExecutor.java

```diff
@@ -186,6 +216,8 @@ private void _walk(PathElement pathElement) {
 		}
 		catch (Throwable throwable) {
 			_log.error(throwable, throwable);
+
+			_completeKaleoInstanceOnFailure(executionContext);
```

Comment:
@pedro-oliveira446 I wouldn't create a new method that it's only used once

Reply:
Done!